File: trustly-android-sdk/src/androidTest/java/net/trustly/android/sdk/views/TrustlyActivityTest.java
package net.trustly.android.sdk.views;

import androidx.test.core.app.ActivityScenario;
import androidx.test.ext.junit.rules.ActivityScenarioRule;

import net.trustly.android.sdk.mock.MockActivity;

import org.junit.After;
import org.junit.Before;
import org.junit.Rule;

public abstract class TrustlyActivityTest {

    @Rule
    public ActivityScenarioRule<MockActivity> activityRule = new ActivityScenarioRule<>(MockActivity.class);

    protected ActivityScenario<MockActivity> scenario;

    @Before
    public void setUp() {
        scenario = activityRule.getScenario();
    }

    @After
    public void tearDown() {
        scenario.close();
    }

}

File: trustly-android-sdk/src/androidTest/java/net/trustly/android/sdk/TrustlyActivityTest.kt
package net.trustly.android.sdk

import androidx.test.core.app.ActivityScenario
import androidx.test.ext.junit.rules.ActivityScenarioRule
import net.trustly.android.sdk.mock.MockActivity
import org.junit.After
import org.junit.Before
import org.junit.Rule
import java.util.concurrent.TimeUnit

abstract class TrustlyActivityTest {

    @JvmField
    @Rule
    var activityRule: ActivityScenarioRule<MockActivity> = ActivityScenarioRule(MockActivity::class.java)

    protected lateinit var scenario: ActivityScenario<MockActivity>

    @Before
    open fun setUp() {
        scenario = activityRule.scenario
    }

    @After
    open fun tearDown() {
        scenario.close()
    }

    open fun waitToCloseCustomTabs() {
        TimeUnit.SECONDS.sleep(2L)
    }

}